src/guiapi/src/gui.cpp

```diff
@@ -67,40 +67,45 @@ void gui_loop(void) {
     uint32_t tick;
 
     #ifdef GUI_JOGWHEEL_SUPPORT
-    Jogwheel::ButtonAction btn = jogwheel.GetButtonAction();
-    bool encoder_changed = jogwheel.EncoderChanged();
-    if (btn == Jogwheel::ButtonAction::BTN_PUSHED) {
+    Jogwheel::BtnState_t btn_ev;
+    volatile bool is_btn = jogwheel.ConsumeButtonEvent(btn_ev);
+    volatile int32_t encoder_diff = jogwheel.GetEncoderDiff();
+
+    // I do not know why this is separate. To play sound asap?
```

Comment:
It was separate because it was developed separately and BTN_PUSHED was a special event, but I think you can add it now to scope ant row:97...

Reply:
fixed